src/main/java/uk/gov/companieshouse/pscstatementdataapi/api/PscStatementApiService.java

```diff
@@ -19,16 +19,27 @@ public class PscStatementApiService {
 
     @Autowired
     InternalApiClient internalApiClient;
+    @Value("${chs.kafka.api.endpoint}")
+    private String chsKafkaApiUrl;
     @Value("${psc-statements.api.resource.changed.uri}")
     private String resourceChangedUri;
     @Value("${psc-statements.api.resource.kind}")
     private String resourceKind;
     private static final String PSC_STATEMENTS_URI = "/company/%s/persons-with-significant-control-statements/%s";
+    private static final String CHANGED_EVENT_TYPE = "changed";
     private static final String DELETE_EVENT_TYPE = "deleted";
     @Autowired
     private Logger logger;
 
+    public ApiResponse<Void> invokeChsKafkaApi(String contextId, String companyNumber, String statementId) {
```

Comment:
is this here for best practice or can the invokeChsKafkaApiWithDeleteEvent not be used for put too

Reply:
The difference is that we pass in a Statement for the Delete method. We could theoretically have only one method and pass in a null Statement for Insert, but it won't be clear why we're doing this in the class that calls the method.
If we stick with two methods, you can tell the function from the method name and we still have similar logic - only one mapChangedResource method - but that's only in private methods within the one class.
Also, this is how other repos have done it.

Reply:
np thanks for clarifying